plugins/modules/identity/keycloak/keycloak_client_scopemapping.py

```diff
@@ -0,0 +1,333 @@
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """
+    Module execution
+
+    :return:
+    """
+    argument_spec = keycloak_argument_spec()
+
+    meta_args = dict(
+        state=dict(default='present', choices=['present', 'absent']),
+        realm=dict(default='master'),
+        client_id=dict(type='str'),
+        id=dict(type='str'),
+        client_role=dict(
+            type='dict',
+            options=dict(
+                client_id=dict(type='str'),
+                id=dict(type='str'),
+                roles=dict(
+                    type='list',
+                    elements='dict',
+                    options=dict(
+                        id=dict(type='str'),
+                        name=dict(type='str')
+                    ),
+                    required_one_of=([['id', 'name']])),
+            ),
+            required_one_of=([['client_id', 'id']])),
+    )
+
+    argument_spec.update(meta_args)
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           required_one_of=([
+                               ['token', 'auth_realm', 'auth_username', 'auth_password'],
+                               ['client_id', 'id'],
+                           ]),
+                           required_together=([
+                               ['auth_realm', 'auth_username', 'auth_password']
+                           ]))
+
+    result = dict(changed=False, msg='', diff={}, proposed={}, existing={}, end_state={})
+
+    # Obtain access token, initialize API
+    try:
+        connection_header = get_token(module.params)
+    except KeycloakError as e:
+        module.fail_json(msg=str(e))
+
+    kc = KeycloakAPI(module, connection_header)
+
+    realm = module.params.get('realm')
+    state = module.params.get('state')
+    client_id = module.params.get('client_id')
+    cid = module.params.get('id')
+    client_role = module.params.get('client_role')
+
+    # Get the potential missing parameters
+    if cid is None:
+        cid = kc.get_client_id(client_id, realm=realm)
+        if cid is None:
+            module.fail_json(msg='Could not fetch client %s:' % client_id)
+
+    if not client_role:
+        module.exit_json(msg="Nothing to do (no roles specified).")
+    if client_role['id'] is None:
+        client_role['id'] = kc.get_client_id(client_role['client_id'], realm=realm)
+        if client_role['id'] is None:
+            module.fail_json(msg='Could not fetch client %s:' % client_role['client_id'])
+    if not client_role['roles']:
+        module.exit_json(msg="Nothing to do (no roles specified).")
+    for role in client_role['roles']:
+        if role['id'] is None:
+            role['id'] = kc.get_client_role_id_by_name(cid, role['name'], realm=realm)
+            if role['id'] is None:
+                module.fail_json(msg='Could not fetch role %s:' % (role['name']))
```

Comment:
Just an observation is that `get_client_role_id_by_name`, even though that function seems to get the full list of client roles and filters for just the name.  This would imply that this solution is wasteful - either the web request could include a query parameter for the role's name, or the list could be filtered for each iteration of this for loop.

However, for most use cases, I don't think this is a great concern.